### ISSUE TITLE
confusing default behavior: calling non-existent method on bean gives no error

### ISSUE THREAD

**Issue by ipsod** [gabordemooij/redbean#872]
This code gives no error:

    R::nuke();
    $f = R::dispense('f');
    $r = $f->hey_there();

It just sets $r to NULL. This is error-prone and makes debugging difficult.  Any other class would give an error.

I had to look hard to find mention of it in the documentation.  It's hidden under the heading "Custom FUSED methods".  And, though I found mention of it, it's not clear what to do next to fix it:

>  If you call a method on a bean that does not exist in the bean and also not in the model the call will be ignored. You change this behaviour by selecting a different FUSE error handling mechanism using the setErrorHandlingFUSE() method, see API. 

So, I'm digging through that now, but... why is this your preferred behavior?  The downside seems obvious.  Is there some benefit?

**Comment by gabordemooij** [comment]
Hi @ipsod,

Too bad you experience some problems while using RedBeanPHP Fuse. Thanks @Lynesth for your detailed explanation. There is not a lot I can add to this explanation. This mechanism has kind of evolved to where it currently is. To be honest (but I am the original author so I am biased) I never have issues with this. Sure, sometimes I forget to hook up the models and things break, but that's almost always only in the beginning of a project. I use RedBeanPHP for all my projects btw. It might also have to do with the way I use RedBeanPHP. If you like you can send me a snippet of your code and we can discuss it. Of course, coding style is a matter of opinion, but RedBeanPHP has been designed with a certain style in mind, so it could be beneficial to see how you use it, and how RedBeanPHP can fit in. Personally, I like to keep things simple. I personally use RedBeanPHP together with my own little micro framework called 'Straight' https://gabordemooij.com/straight/ - but that is certainly not for everyone. Anyway, wish you good luck and thanks for discussing this with us.

cheers,
Gabor

**Comment by ipsod** [comment]
Hi @gabordemooij ,

I had decided yesterday to try "Straight" - I wanted a framework, and it is so simple.  Thank you for your offer to discuss my code - I hope to take you up on it soon!

I'm not having this trouble any more, since I'm now using models everywhere, but I do have other troubles.

Is there a preferred place to ask questions?  Also, is there a place to find example projects that use best-practices?

Just last night I started writing my first complex form, and it is a job that would have taken less than 10 minutes and very few lines of code in Django to make the route, validation, form with CSRF protection and validation errors, and save-to-database, and these things are almost fully automatic, so that if I updated my Django model these things would automatically reflect that update.  It's taken me much longer with PHP, and so many more lines of code, and I am still not done.  I'm sure I'm addressing the problem at too low a level, but I don't think I want to reinvent the wheel by writing my own validation library or trying to figure out how to tie these things (validation, model population, form creation with showing errors) myself.  I would really appreciate if there were examples or docs to provide guidance.
